### PR TITLE
Migrate `run-make/return-non-c-like-enum` to `rmake.rs`

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/rust-lang/rust/issues/121876.

r? @Kobzol

### CODE REVIEW

File: tests/run-make/return-non-c-like-enum/rmake.rs
Comment (jieyouxu) [inline tests/run-make/return-non-c-like-enum/rmake.rs:1]: Suggestion: could we add a test comment for this? Something like

> Check that we treat enum variants like union members in call ABIs.
> Added in <https://github.com/rust-lang/rust/pull/68443>.
> Original issue: <https://github.com/rust-lang/rust/issues/68190>.

**Review by jieyouxu — APPROVED** [review]
Thanks, just one suggestion for adding a test comment, then r=me.

**Comment by Kobzol** [comment]
You can r=me with the comment added (in general, it's great to add comments to run-make tests that try to explain at least a little bit what the test is about, or link to existing issues etc.).

**Comment by GuillaumeGomez** [comment]
So I also somewhow removed the wrong file in the tidy list. So I fixed that and added the comment for what the test is about. :)

**Pull request merged**